src/pages/WizzardPage3/WizzardPage3.js

```diff
@@ -0,0 +1,509 @@
+import useInput from "../../hooks/useInput";
+import WizzardButtons from "../../components/Wizzard/WizzardButtons/WizzardButtons";
+import ConvictedDetails from "../../components/Wizzard/ConvictedDetails/ConvictedDetails";
+import { Contex } from "../../store/Store";
+
+const WizzardPage3 = (props) => {
+  const { onForm3Submit, setInWizzard, setStep } = props;
+
+  const ctx = useContext(Contex);
+  const wizardData = ctx.state.applicationInfo;
+
+  const navigate = useNavigate();
+
+  const [state, setState] = useState({
+    agriculturalSkills: wizardData.agriculturalSkills,
+    agriculturalSkillsDetails: wizardData.agriculturalSkillsDetails,
+    metalworkSkills: wizardData.metalworkSkills,
+    metalworkSkillsDetails: wizardData.metalworkSkillsDetails,
+    Marking: wizardData.Marking,
+    markingValue: wizardData.markingValue,
+    Cutting: wizardData.Cutting,
+    cuttingValue: wizardData.cuttingValue,
+    Drilling: wizardData.Drilling,
+    drillingValue: wizardData.drillingValue,
+    CuttingThreads: wizardData.CuttingThreads,
+    cuttingThreadsValue: wizardData.cuttingThreadsValue,
+    Filling: wizardData.Filling,
+    fillingValue: wizardData.fillingValue,
+    Joining: wizardData.Joining,
+    joiningValue: wizardData.joiningValue,
+    convicted: wizardData.convicted,
+    reasons: wizardData.reasons,
+    errorMessageReasons: [],
+    airplaneSkills: wizardData.airplaneSkills,
+    carSkills: wizardData.carSkills,
+    bicycleSkills: wizardData.bicycleSkills,
+  });
+  console.log(wizardData);
+
+  const [formIsValid, setFormIsValid] = useState(true);
+
+  const agriculturalSkillsInputHandler = (e) => {
+    setState((prev) => ({ ...prev, agriculturalSkills: e.target.value }));
+
+    if (e.target.value === "No") {
+      setState((prev) => ({ ...prev, agriculturalSkills: "No" }));
+      setState((prev) => ({ ...prev, agriculturalSkillsDetails: "" }));
+    }
+  };
+
+  const {
+    isYes: affirmativeAnswerAgricultural,
+    isRadioSelected: isRadioSelectedAgricultural,
+    handleRadioClick: handleRadioClickAgricultural,
+  } = useRadio(
+    (val) => {
+      return val === "Yes";
+    },
+    agriculturalSkillsInputHandler,
+    state.agriculturalSkills
+  );
+
+  const agriculturalSkillsDetailsInputHandler = (e) => {
+    setState((prev) => ({
+      ...prev,
+      agriculturalSkillsDetails: e.target.value,
+    }));
+  };
+
+  const {
+    onChangeFunc: onAgriculturalSkillsDetailsChangeFunc,
+    onBlurFunc: onAgriculturalSkillsDetailsBlurFunc,
+    isInvalid: agriculturalSkillsDetailsInvalid,
+  } = useInput(
+    (val) => {
+      return val.trim() !== "" && val.length < 50;
+    },
+    agriculturalSkillsDetailsInputHandler,
+    state.agriculturalSkillsDetails
+  );
+
+  const metalworkSkillsInputHandler = (e) => {
+    setState((prev) => ({ ...prev, metalworkSkills: e.target.value }));
+
+    if (e.target.value === "No") {
+      setState((prev) => ({ ...prev, Marking: false }));
```

Comment:
You should not revert these states manually, you should just set the initial state, and revert them to that initial state.